### PR TITLE
Add Approval API docs for PAG

### PR DESCRIPTION
**Pull request by mandardalvi**
This PR is tracked via this work item https://dev.azure.com/msft-skilling/Content/_queries/edit/95726/?triage=true

### CODE REVIEW

File: api-reference/beta/api/approval-list-steps.md [hunk 1]
```diff
@@ -15,24 +15,42 @@ Namespace: microsoft.graph
 
 In [Azure AD entitlement management](../resources/entitlementmanagement-overview.md), lists the [approvalStep](../resources/approvalstep.md) objects associated with an [approval](../resources/approval.md) object.  This call can be made by an approver, providing the identifier of the [access package assignment request](../resources/accesspackageassignmentrequest.md).
 
+In [PIM for groups](../resources/privilegedidentitymanagement-for-groups-api-overview.md), lists the [approvalStep](../resources/approvalstep.md) objects associated with an [approval](../resources/approval.md) object.  This call can be made by an approver, providing the identifier of the [assignment schedule request](../resources/privilegedaccessgroupassignmentschedulerequest.md).
+
```
Comment (FaithOmbongi) [inline api-reference/beta/api/approval-list-steps.md:19]:
```suggestion
List the [approvalStep](../resources/approvalstep.md) objects associated with an [approval](../resources/approval.md). This API request is made by an approver in the following scenarios:

- In [entitlement management](../resources/entitlementmanagement-overview.md), providing the identifier of the [access package assignment request](../resources/accesspackageassignmentrequest.md).
- In [PIM for groups](../resources/privilegedidentitymanagement-for-groups-api-overview.md), providing the identifier of the [assignment schedule request](../resources/privilegedaccessgroupassignmentschedulerequest.md).

```

File: api-reference/beta/api/approval-list-steps.md [hunk 2]
```diff
@@ -15,24 +15,42 @@ Namespace: microsoft.graph
...
 ## Permissions
 
 One of the following permissions is required to call this API. To learn more, including how to choose permissions, see [Permissions](/graph/permissions-reference).
 
+### Permissions required for calling this API for entitlement management
```
Comment (FaithOmbongi) [inline api-reference/beta/api/approval-list-steps.md:24]:
```suggestion
### For entitlement management
```

File: api-reference/beta/api/approval-list-steps.md [hunk 3]
```diff
@@ -15,24 +15,42 @@ Namespace: microsoft.graph
...
+
 | Permission type                        | Permissions (from least to most privileged) |
 |:---------------------------------------|:--------------------------------------------|
 | Delegated (work or school account)     | EntitlementManagement.Read.All, EntitlementManagement.ReadWrite.All |
 | Delegated (personal Microsoft account) | Not supported. |
 | Application                            | Not supported. |
 
+### Permissions required for calling this API for PIM for groups
```
Comment (FaithOmbongi) [inline api-reference/beta/api/approval-list-steps.md:32]:
```suggestion
### For PIM for groups
```

File: api-reference/beta/api/approval-list-steps.md [hunk 4]
```diff
@@ -15,24 +15,42 @@ Namespace: microsoft.graph
...
 ```
 
+### List approval steps example for PIM for groups.
+```http
+GET /identityGovernance/privilegedAccess/group/assignmentApprovals/{id}/steps
+```
+
 ## Request headers
 
 | Name      |Description|
```
Comment (FaithOmbongi) [inline api-reference/beta/api/approval-list-steps.md:56]:
```suggestion
## HTTP request

To list the approval steps in entitlement management:

<!-- { "blockType": "ignored" } -->
```http
GET /identityGovernance/entitlementManagement/accessPackageAssignmentApprovals/{id}/steps
```

To list the approval steps in PIM for groups:


```http
GET /identityGovernance/privilegedAccess/group/assignmentApprovals/{id}/steps
```

## Request headers

| Name      |Description|
```

File: api-reference/beta/api/approval-list-steps.md [hunk 5]
```diff
@@ -60,10 +78,16 @@ The following is an example of the request.
...
 GET https://graph.microsoft.com/beta/identityGovernance/entitlementManagement/accessPackageAssignmentApprovals/abd306ef-f7b2-4a10-9fd1-493454322489/steps
 ```
 
+### Example 2: List approval steps example for PIM for groups.
+```msgraph-interactive
+GET https://graph.microsoft.com/beta/identityGovernance/privilegedAccess/group/assignmentApprovals/abd306ef-f7b2-4a10-9fd1-493454322489/steps
+```
+
 # [C#](#tab/csharp)
 [!INCLUDE [sample-code](../includes/snippets/csharp/get-approvalstep-1-csharp-snippets.md)]
```
Comment (FaithOmbongi) [inline api-reference/beta/api/approval-list-steps.md:92]: 1 request to 1 response. You can keep the existing entitlement management example only or add an example 2 with request: response for PIM for groups.
```suggestion

```http
GET https://graph.microsoft.com/beta/identityGovernance/entitlementManagement/accessPackageAssignmentApprovals/abd306ef-f7b2-4a10-9fd1-493454322489/steps
```

# [C#](#tab/csharp)
[!INCLUDE [sample-code](../includes/snippets/csharp/get-approvalstep-1-csharp-snippets.md)]
```

File: api-reference/beta/api/approvalstep-get.md
Comment (FaithOmbongi) [inline api-reference/beta/api/approvalstep-get.md:1]: All the comments in the other doc apply here

File: api-reference/beta/api/approvalstep-update.md
Comment (FaithOmbongi) [inline api-reference/beta/api/approvalstep-update.md:1]: All the comments in the other doc apply here.

**Review by FaithOmbongi — CHANGES_REQUESTED** [review]
Hi @mandardalvi , @ilyalushnikov - I've left feedback inline...please address them.

Also, document assignmentApprovals relationship in privilegedAccessGroup.md

Q: Aren't you documenting these APIs for PIM for Azure AD roles as well? 

I'm curious, how come v1 PIM for Azure AD roles doesn't have these approval capabilities? i.e. there's no approval relationship in the schema. Was this an oversight in the schema publication?

File: api-reference/beta/resources/privilegedidentitymanagement-for-groups-api-overview.md [hunk 6]
```diff
@@ -32,8 +32,10 @@ The following table lists scenarios for using PIM for groups APIs to manage acti
...
-| A principal queries their membership or ownership assignment requests and the details | [privilegedAccessGroupAssignmentScheduleRequest: filterByCurrentUser](../api/privilegedaccessgroupassignmentschedulerequest-filterbycurrentuser.md) |
+| A principal queries their membership or ownership assignment requests and the details <br/><br/> An approver queries membership or ownership requests waiting for their approval and details of these requests | [privilegedAccessGroupAssignmentScheduleRequest: filterByCurrentUser](../api/privilegedaccessgroupassignmentschedulerequest-filterbycurrentuser.md) |
 | A principal cancels a membership or ownership assignment request they created | [privilegedAccessGroupAssignmentScheduleRequest: cancel](../api/privilegedaccessgroupassignmentschedulerequest-cancel.md) |
+| An approver gets details for approval request, including information about approval steps | [Get approval](../api/approval-get.md) |
+| An approver approves or denies approval request by approving or denying approval step | [Update approvalStep](../api/approvalstep-get.md) |
```
[the remark this answers is not in the document]
Comment (ilyalushnikov) [reply api-reference/beta/resources/privilegedidentitymanagement-for-groups-api-overview.md:38]: "the steps to approve the request" isn't accurate IMHO. "steps to approve the request" gives an impression that these are steps that approver has to take. Which isn't true in this situation.
Can we keep "information about approval steps" in line 37 and implement rest of your suggestions?

File: api-reference/beta/resources/privilegedidentitymanagement-for-groups-api-overview.md [hunk 7]
```diff
@@ -83,9 +85,12 @@ The following Microsoft Graph permissions are required to call the PIM for group
 | assignmentScheduleRequest | CREATE, LIST, GET, UPDATE, DELELE | PrivilegedAssignmentSchedule.ReadWrite.AzureADGroup |
 | eligibilitySchedule <br/> eligibilityScheduleInstance | LIST, GET | PrivilegedEligibilitySchedule.Read.AzureADGroup <br/> PrivilegedEligibilitySchedule.ReadWrite.AzureADGroup |
 | eligibilityScheduleRequest | CREATE, LIST, GET, UPDATE, DELELE | PrivilegedEligibilitySchedule.ReadWrite.AzureADGroup |
+| approval | GET | PrivilegedAssignmentSchedule.Read.AzureADGroup <br/> PrivilegedAssignmentSchedule.ReadWrite.AzureADGroup |
```
[the remark this answers is not in the document]
Comment (ilyalushnikov) [reply api-reference/beta/resources/privilegedidentitymanagement-for-groups-api-overview.md:88]: In my opinion, this overview is incredibly useful for a reader.
I may have different background though: I didn't work on API documentation ever before, but I did publish a book and delivered hundreds of technical trainings. Documentation may be more dry and avoid repeating information, while in the book or working with live audience you go from broader strokes and get into details later. That's the reason I would like to keep this section if it doesn't break any rules.

File: api-reference/beta/resources/privilegedidentitymanagement-for-groups-api-overview.md (hunk 6 above)
[the remark this answers is not in the document]
Comment (FaithOmbongi) [reply api-reference/beta/resources/privilegedidentitymanagement-for-groups-api-overview.md:38]: Ah, OK. No problem. I defer to you for technical accuracy.

File: api-reference/beta/resources/privilegedidentitymanagement-for-groups-api-overview.md (hunk 7 above)
[the remark this answers is not in the document]
Comment (FaithOmbongi) [reply api-reference/beta/resources/privilegedidentitymanagement-for-groups-api-overview.md:88]: Doesn't break any rules. If readers find this summary useful, we can continue to keep it.

File: api-reference/beta/api/approval-get.md [hunk 8]
```diff
@@ -13,26 +13,45 @@ Namespace: microsoft.graph
...
 
 <!-- { "blockType": "ignored" } -->
 
+### To get approval example for entitlement management.
 ```http
 GET /identityGovernance/entitlementManagement/accessPackageAssignmentApprovals/{id}
 ```
 
+### To get approval example for PIM for groups.
+```http
```
Comment (FaithOmbongi) [inline api-reference/beta/api/approval-get.md:51]:
```suggestion
To list the approval steps in entitlement management.
```http
GET /identityGovernance/entitlementManagement/accessPackageAssignmentApprovals/{id}
```

To list the approval steps in PIM for groups:
```http
```

**Review by FaithOmbongi — APPROVED** [review]
Pushed a few commits to fix the structure of the HTTP request syntax section then approved.

cc @mandardalvi, @ilyalushnikov

**Pull request merged**